code/game/objects/items/weapons/tools/_tools.dm

```diff
@@ -681,6 +681,13 @@
 	return return_quality
 
 /obj/item/weapon/tool/proc/turn_on(var/mob/user)
+	if(!cell)
```

Comment:
You have surrond with a if block if the tool actually requires to have a cell when the tool is using fuel and the turn_on() calls the ..()